src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -336,7 +336,9 @@ private async Task<DebugLaunchSettings> GetConsoleTargetForProfile(ILaunchProfil
             bool useCmdShell = false;
             if (await IsConsoleAppAsync())
             {
-                if (await IsIntegratedConsoleEnabledAsync())
+                var isIntegratedConsoleCapable = _project.Capabilities.Contains(ProjectCapabilities.IntegratedConsoleDebugging);
```

Comment:
Your first guess was on the right track, `ConsoleDebugTargetsProvider` lives in the `ConfiguredProject` scope because it [exports this](https://github.com/dotnet/project-system/blob/02514b0f9ac016b6ecc19e1fb93eed9646f0ee87/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/IDebugProfileLaunchTargetsProvider.cs#L15) that means that this capabilities check should be based on `ConfiguredProject.Capabilities`.

You can get your hands on a `ConfiguredProject`, just by adding another argument `ConfiguredProject configuredProject` to the constructor

Reply:
Thanks, I've addressed that now